src/raftstore/store/peer.rs

```diff
@@ -1471,14 +1478,14 @@ impl Peer {
         // guarantee the orders are consistent with read_states. `advance` will
         // update the `read_index` of read request that before this successful
         // `ready`.
-        if !self.is_leader() && !ready.read_states().is_empty() {
+        if !self.is_leader() {
             // NOTE: there could still be some read requests following, which will be cleared in
             // `clear_uncommitted` later.
             for state in ready.read_states() {
                 self.pending_reads
                     .advance(state.request_ctx.as_slice(), state.index);
-                self.post_pending_read_index_on_replica(ctx);
             }
+            self.post_pending_read_index_on_replica(ctx);
```

Comment:
why calling it even `read_states` is empty?

Reply:
For handling the delayed requests.